src/components/Button.js

```diff
@@ -34,7 +39,13 @@ class Button extends React.Component<Props> {
   };
 
   render() {
-    return <pin pin={this.props.pin} onRead={this.onRead} mode={'INPUT'} />;
+    return (
+      <pin
+        pin={this.props.pin}
+        onRead={this.onRead}
+        mode={this.props.internalPullup ? 'PULLUP' : 'INPUT'}
```

Comment:
Do you think this optional boolean prop makes more sense than having the prop be `mode?: 'PULLUP' | 'INPUT'` with `INPUT` being the default if mode is anything except `PULLUP`? (naive question, I literally have no hardware experience)

Reply:
> I literally have no hardware experience

Full disclosure, I don't either. I've done a few small things with my Arduino but not much. 

Regardless, I had gathered the intent of these components to be wrappers for direct `pin` component references. This pattern pushed me away from forcing implementers to use a string for mode selection and pushing them to use a simple boolean property. 

Basically, the intent was that the implementer just needs to say they want to use the internal pullup - not learn the (admittedly clear) enums that Firmata uses internally.

Reply:
Yeah, this sounds sound enough for me. Thanks!